crates/subspace-farmer/src/single_disk_farm.rs

```diff
@@ -655,6 +655,98 @@ impl ScrubTarget {
     }
 }
 
+struct AllocatedSpaceDistribution {
+    piece_cache_file_size: u64,
+    piece_cache_capacity: u32,
+    plot_file_size: u64,
+    target_sector_count: u16,
+    metadata_file_size: u64,
+}
+
+impl AllocatedSpaceDistribution {
+    fn new(
+        allocated_space: u64,
+        sector_size: u64,
+        cache_percentage: u8,
+        sector_metadata_size: u64,
+    ) -> Result<Self, SingleDiskFarmError> {
+        let single_sector_overhead = sector_size + sector_metadata_size;
+        // Fixed space usage regardless of plot size
+        let fixed_space_usage = RESERVED_PLOT_METADATA
+            + RESERVED_FARM_INFO
+            + Identity::file_size() as u64
+            + KnownPeersManager::file_size(KNOWN_PEERS_CACHE_SIZE) as u64;
+        // Calculate how many sectors can fit
+        let target_sector_count = {
+            let potentially_plottable_space = allocated_space.saturating_sub(fixed_space_usage)
+                / 100
+                * (100 - u64::from(cache_percentage));
```

Comment:
Do we have a "non-100%" check somewhere?

Reply:
We have in CLI. If someone specifies 100 as an input here it'll simply return wrong answer to the wrong input, nothing criminal.